include/private/node/nettest_wrap.hpp

```diff
@@ -306,6 +307,27 @@ template <typename Nettest> class NettestWrap : public Nan::ObjectWrap {
         });
     }
 
+    /// The on_overall_data_usage setter allows to set the callback called when
+    /// the overall data used by the test is available.
+    static void on_overall_data_usage(const Nan::FunctionCallbackInfo<v8::Value> &info) {
+        set_value(1, info, [&info](NettestWrap *self) {
+            self->nettest.on_overall_data_usage([
+                async_ctx = self->async_ctx, callback = wrap_callback(info[0])
+            ](DataUsage du) {
+                UvAsyncCtx<>::suspend(async_ctx, [callback, du]() {
+                    Nan::HandleScope scope;
+                    const int argc = 2;
+                    // XXX can downcasting uint64_t to uint32_t create issues?
```

Comment:
This is done because the `Nan` template doesn't have a prototype for `uint64_t`, but does have one for `uint32_t`.

Reply:
does it have one for `int64_t`?

Reply:
I would say that downcasting is really really bad here.